sdk/python/lib/pulumi/dynamic/dynamic.py

```diff
@@ -234,6 +234,9 @@ class Resource(CustomResource):
     Resource represents a Pulumi Resource that incorporates an inline implementation of the Resource's CRUD operations.
     """
 
+    def __init_subclass__(cls, name: str ='dynamic:Resource'):
+        cls._resource_type_name = name
+
```

Comment:
I'd prefer that we keep the `dynamic` portion of the type token. Removing that portion of the token makes it dangerous for us to add any additional builtin resources to the `pulumi-python` provider. If you want to change the module, we could instead do something like this:

```suggestion
    def __init_subclass__(cls, module: str = '', name: str ='Resource'):
        if module != '':
            module = '/' + module
        cls._resource_type_name = f"dynamic{module}:{name}"

```

This would let you form type tokens like `pulumi-python:dynamic/my-module:Resource`. The `pulumi-python` provider would then be able to safely add other resources outside of the `dynamic` namespace in the future.